README.md

```diff
@@ -1,2 +1,148 @@
 # terraform-provider-jupiterone
-Terraform provider for JupiterOne
+
+**NOTE: This project is currently in beta and is _not_ ready for production use.**
+
+Requirements
+------------
+
+-	[Terraform](https://www.terraform.io/downloads.html) 0.10.x
+-	[Go](https://golang.org/doc/install) 1.11 (to build the provider plugin)
+
+## Example Usage
+
+```terraform
+# Configure the JupiterOne provider
+provider "jupiterone" {
+  api_key = "${var.jupiterone_api_key}"
+  account_id = "${var.jupiterone_account}"
+}
+
+# Create a new JupiterOne rule
+resource "jupiterone_rule" "unencrypted_critical_data_stores" {
+  # ...
+}
+```
+
+Building The Provider
+---------------------
+
+Clone repository to: `$GOPATH/src/github.com/jupiterone/terraform-provider-jupiterone`
+
+```sh
+$ mkdir -p $GOPATH/src/github.com/jupiterone; cd $GOPATH/src/github.com/jupiterone
+$ git clone git@github.com:jupiterone/terraform-provider-jupiterone
+```
+
+Enter the provider directory and build the provider
+
+```sh
+$ cd $GOPATH/src/github.com/jupiterone/terraform-provider-jupiterone
+$ make build
+```
+
+Using the provider
+----------------------
+If you're building the provider, follow the instructions to [install it as a plugin.](https://www.terraform.io/docs/plugins/basics.html#installing-a-plugin) After placing it into your plugins directory,  run `terraform init` to initialize it.
+
+Developing the Provider
+---------------------
+
+If you wish to work on the provider, you'll first need [Go](http://www.golang.org) installed on your machine (please check the [requirements](https://github.com/jupiterone/terraform-provider-jupiterone#requirements) before proceeding). You'll also need to correctly setup a [GOPATH](http://golang.org/doc/code.html#GOPATH), as well as adding `$GOPATH/bin` to your `$PATH`.
+
+*Note:* This project uses [Go Modules](https://blog.golang.org/using-go-modules) making it safe to work with it outside of your existing [GOPATH](http://golang.org/doc/code.html#GOPATH).
+
+To compile the provider, run `make build`. This will build the provider and put the provider binary in the `$GOPATH/bin` directory.
+
+```sh
+$ make build
+...
+$ $GOPATH/bin/terraform-provider-jupiterone
+...
+```
+
+In order to test the provider, you can simply run `make test`. Pre-recorded API responses
+(cassettes) are run. The cassettes are stored in `jupiterone/cassettes/`.
+When tests are modified, the cassettes need to be re-recorded.
```

Comment:
Neat

Reply:
Yah, love this. I think we need to promote similar testing of integrations. I'm using Polly.js for this purpose.

Reply:
Ya I adapted the approach that the datadog provider was taking. I think it's awesome.

It's also worth noting that noisy headers and auth-related headers are stripped out before creating the cassette here: https://github.com/JupiterOne/terraform-provider-jupiterone/pull/1/files#diff-42149ce5006a5ff05ad7272250b5c53eR73

Reply:
I purposely left the `X-Amzn-Requestid` (and a few others) that I felt may be helpful for debugging contributor code.

Reply:
Yeah, I'm think we could probably add some testing utilities to the SDK for setting up Polly with some sane defaults and ofc make it extensible so that provider specific fields/headers can be redacted.